tests/helpers.py

```diff
@@ -33,9 +33,9 @@ def getkey():
 
 
 @ft.lru_cache(maxsize=None)
-def _construct_matrix_impl(getkey, cond_cutoff, tags, size):
+def _construct_matrix_impl(getkey, cond_cutoff, tags, size, dtype=jnp.float64):
```

Comment:
I'd suggest removing the default, and just explicitly specifying it everywhere. Default arguments are often a source of unexpected behaviour, so my usual rule is to avoid them across internal APIs.